retwist/util/sentry.py

```diff
@@ -1,19 +1,29 @@
+from typing import Dict
+
 import sentry_sdk
 from twisted.python import log
 from twisted.python.failure import Failure
 from twisted.web.http import Request
 
 
-def add_request_context_to_scope(request, scope):
-    # type: (Request, sentry_sdk.Scope) -> None
+def add_request_context_to_scope(request, scope, context):
+    # type: (Request, sentry_sdk.Scope, Dict) -> None
     """
-    Add context information which is useful for debugging from a Twisted request to a Sentry scope.
+    Add error context information which is useful for debugging from a Twisted request to a Sentry scope.
     """
-    scope.set_extra("url", request.uri)
-    scope.set_extra("method", request.method)
-    scope.set_extra("headers", request.getAllHeaders())
-    scope.set_extra("query_string", request.uri.replace(request.path, b"").strip(b"?"))
-    scope.set_extra("data", request.args)
+    request_context = {
+        "url": request.uri,
+        "method": request.method,
+        "headers": request.getAllHeaders(),
+        "query_string": request.uri.replace(request.path, b"").strip(b"?"),
+        "data": request.args
+    }
+
+    if "user_id" in context:
+        scope.set_user({"id": context.pop("user_id", None)})
+
+    request_context.update(context)
+    scope.set_context("request", request_context)
```

Comment:
I see you've changed the `sentry_sdk` methods you use, `set_user` and `set_context` instead of `set_extra`, but checking the docs this seems to be proper use, and I'm OK with this change.

Reply:
I have been reading about a deprecation of set_extra in https://docs.sentry.io/platforms/python/enriching-events/context/#additional-data. Basically my intention in the code was to avoid the deprecated features and use what is recommended.